src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -181,10 +208,21 @@ private unsafe int ReadFromConnection(void* connection, byte* data, void** dataL
                 return noErr;
             }
 
+            if (Interlocked.Exchange(ref _pendingInput, NativeIOPending) == Disposed)
+            {
+               const int readErr = -19;
+               return readErr;
+            }
+
             uint transferred = 0;
 
             if (_inputBuffer.ActiveLength == 0)
             {
+                if (Interlocked.Exchange(ref _pendingInput, NoIOPending) == Disposed)
```

Comment:
Should this use CompareExchange?

I'll stop commenting on these.  Basically, all the ones that are comparing against Disposed seem... suspicious.